motors_weg_cvw300Types.hpp

```diff
@@ -1,14 +1,26 @@
 #ifndef motors_weg_cvw300_TYPES_HPP
 #define motors_weg_cvw300_TYPES_HPP
 
-/* If you need to define types specific to your oroGen components, define them
- * here. Required headers must be included explicitly
- *
- * However, it is common that you will only import types from your library, in
- * which case you do not need this file
- */
+#include <base/Time.hpp>
+#include <motors_weg_cvw300/Configuration.hpp>
+#include <motors_weg_cvw300/InverterStatus.hpp>
 
 namespace motors_weg_cvw300 {
+    struct SerialWatchdog {
```

Comment:
Shouldn't this be in the configuration namespace?

Reply:
The story of the configuration namespace:

- I wanted to put configuration datatypes in a single file
- I wanted to keep the file <> name correspondance

Hence, I put all the types in a separate namespace... But then I'm not so happy with the result. Would you have a suggestion ?

Reply:
Not sure I understand.. How do you keep file <> name correspondance when you have more than one type in the same file? All configurations in the same file and in the same namespace sound like what we should be trying to enforce whenever possible IMO

Reply:
> How do you keep file <> name correspondance when you have more than one type in the same file?

As an alternative, to avoid having 1000 files, I've been often using 1 file == 1 namespace when it makes sense. That's what I've done here.

> All configurations in the same file and in the same namespace sound like what we should be trying to enforce whenever possible IMO

Yeah ... I'm not so sure about that. I kind of feel that it would create too many namespaces.

Reply:
I meant always in the `configuration` namespace

Reply:
So ... moved it to the `configuration` namespace for consistency.

I think we should have a longer discussion on how to structure future drivers w.r.t. to this, as well as naming. I'm still not sold on having a `configuration` namespace to be honest.

Reply:
I agree.. Also, given the absolute lack of a decent linter, I'm all for writing a section about coding style and conventions in https://www.rock-robotics.org/rock-and-syskit/